tests/baselines/reference/sourceMapValidationDecorators.sourcemap.txt

```diff
@@ -182,9 +182,8 @@ sourceFile:sourceMapValidationDecorators.ts
 >>>    Greeter.prototype.fn = function (x) {
 1->^^^^
 2 >    ^^^^^^^^^^^^^^^^^^^^
-3 >                        ^^^
-4 >                           ^^^^^^^^^^
-5 >                                     ^
+3 >                        ^^^^^^^^^^^^^

```

Comment:
This shouldnt have changed right? i mean we do want `=` as separate thing in sourcemap ?

Reply:
This is a consequence of performing transforms in two steps where we first remove the modifiers (in the `ts` transform), and *then* remove the decorators (in the `legacyDecorators` transform). It's not the `=` that is mapped here, but rather the start of the function expression that is no longer mapped to the start of the `private` modifier since that modifier no longer exists when the decorators transform is run. I debated accepting this change, but modifiers aren't a valid position for an inline breakpoint so this doesn't seem to affect the debug experience.

Reply:
It turns out the information *is* available, since the `decorators` and `modifiers` arrays exist and have a pos/end but are empty. The issue is that `moveRangePastDecorators` and `moveRangePastModifiers` only use the `end` of a `NodeArray` if the array isn't empty. I may be able to change that to get the necessary information to insert a valid source map position here.

Reply:
I've pushed up a change to address this. There are still changes to the sourcemap emit, but these changes should be an improvement as the mappings are more consistent.